File: gcc/testsuite/gfortran.dg/reassoc_5.f90
! { dg-do compile }
! { dg-options "-O3 -ffast-math -fdump-tree-optimized -fno-protect-parens" }
!
! PR fortran/35259
! Test for -fno-protect-parens
!
function test(b)
  real a
  a = (b + 5.) - 5.
  test = a
end

! Test copied from reassoc_1.f90 which checked for -fprotect-parens (default),
! and thus for the occurance of "5 - 5".
!
! We need an explicit +5 and -5, and an intermediate ((bla)) expression
! (the reassoc barrier).  Make use of "." matching lineends.
! { dg-final { scan-tree-dump-times "\\\+ 5.*\\\)\\\).* - 5" 0 "optimized" } }

File: gcc/testsuite/gcc.dg/tree-ssa/reassoc-5.c
/* { dg-do compile } */ 
/* { dg-options "-O2 -fdump-tree-optimized" } */
extern int printf (const char *, ...);
int main(int argc, int b)
{
  /* We should be able to get rid of the a - i.  */
  int i;
  for (i = 0; i < 50; i++)
    {
      int a = b + i;
      int c = a - i;
      int d = argc + b;
      printf ("%d %d\n", c,d);
    }
}
/* { dg-final { scan-tree-dump-times "a - i" 0 "optimized"} } */
